packages/remix-react/components.tsx

```diff
@@ -1138,7 +1138,7 @@ export function useSubmitImpl(key?: string): SubmitFunction {
 
         // Include name + value from a <button>
         if (target.name) {
-          formData.set(target.name, target.value);
+          formData.append(target.name, target.value);
```

Comment:
I'm going to swap this temporarily just to make sure the tests fail without this change. Thanks for your patience :)

```suggestion
          formData.set(target.name, target.value);
```